engineering/core/platform_withdrawl_expiration.md

```diff
@@ -0,0 +1,27 @@
+# Platform Withdrawl Expiration
+Date: 2024-01-31
+Participants: Ody, shumkov, knst, quantumexplorer, Pasta, Igor Markin
+Scope(s): Core, Platform
+
+## Status
+Proposed
+
+## Context
+The handling of asset unlock transactions in the Dash core code has been a subject of discussion. Initially, it was thought that asset unlock transactions should not expire, aligning with the perspective that if a transaction is valid and complies with consensus rules, it should be considered valid indefinitely. However, concerns were raised regarding the complexity and security implications of not having an expiration for these transactions. Specifically, it was debated whether a transaction, once signed, should be valid indefinitely or if it should expire after a certain period or under specific conditions.
+
+## Decision
+It has been decided that asset unlock transactions will expire in core 48 blocks after the signing height (requestedHeight). The platform may reissue these transactions with the same index in the case of expiration. This decision is driven by the need to manage transaction validity efficiently and to ensure the security and integrity of the transaction process. By having transactions expire, it prevents the indefinite validity of transactions which could lead to potential security risks or complications in transaction management. The reissuing mechanism by the platform ensures that transactions that are still valid from a business logic perspective can be reattempted without changing their fundamental identity, as indicated by the index.
```

Comment:
This sounds too general. I can't confirm this without details and an understanding of the process. 

> The platform may reissue these transactions with the same index
This statement is controversial. We use the transaction index for signature request ID so will it be reused for different transactions to sign intentionally? Transactions will differ since we need to set a new quorum hash and sign height which makes it a new transaction.

Reply:
Yeah; this is correct. We won't be able to "resign" as the requestID is only the index. 

An option to solve this is to change the requestID to be index and sign height. I don't think this would need to be a breaking change.

Another option is that platform wouldn't be able to "resign" until the old individual signatures expire which I think is about 2 weeks

Reply:
Tenderdash is able to resign because it passes "submit=false" and it works fine, @lklimek says they already use this feature widely and it is well tested